Sources/Quick/QuickTestExamplesLoader.swift

```diff
@@ -0,0 +1,76 @@
+#if !SWIFT_PACKAGE
+
+import Foundation
+
+/// Singleton which can call `buildExamplesIfNeeded` on each `QuickSpec` subclass once per lifetime of application
+@objc internal final class QuickTestExamplesLoader: NSObject {
```

Comment:
Note that this class consists of 99% extracted logic from QuickTestObservation
The only thing changed - I've added a flag `didBuildExamples`

Reply:
I think this PR will have a better chance of getting reviewed and merged more quickly by maintainers if you remove all the refactoring and keep the diff as small as possible.

The refactoring (which I agree with) can come in a 2nd PR after the main fix is merged. 😊

Reply:
Hm, good idea
Created another one: #1129